### ISSUE TITLE
Really slow startup times

### ISSUE THREAD

**Issue by sudokit** [Blacksuan19/init.nvim#21]
I removed all previous configs and did a clean install on this. Here's the output of: `nvim --startuptime vim.log` and `cat vim.log`:

`times in msec
 clock   self+sourced   self:  sourced script
 clock   elapsed:              other lines

000.007  000.007: --- NVIM STARTING ---
000.284  000.277: locale set
000.589  000.304: inits 1
000.599  000.010: window checked
000.601  000.002: parsing arguments
002.617  002.016: init lua interpreter
002.743  000.126: expanding arguments
002.768  000.025: inits 2
003.158  000.390: init highlight
003.159  000.002: waiting for UI
003.931  000.771: done waiting for UI
003.943  000.012: init screen for UI
003.955  000.012: init default mappings
003.975  000.020: init default autocommands
004.269  000.033  000.033: sourcing /usr/share/nvim/runtime/ftplugin.vim
004.300  000.016  000.016: sourcing /usr/share/nvim/runtime/indent.vim
004.327  000.007  000.007: sourcing /usr/share/nvim/archlinux.vim
004.329  000.019  000.012: sourcing /etc/xdg/nvim/sysinit.vim
005.375  000.791  000.791: sourcing /home/sudokit/.config/nvim/autoload/plug.vim
008.255  000.040  000.040: sourcing /usr/share/nvim/runtime/filetype.lua
012.074  000.007  000.007: sourcing /home/sudokit/.config/nvim/plugged/vim-fugitive/ftdetect/fugitive.vim
012.132  000.019  000.019: sourcing /home/sudokit/.config/nvim/plugged/cypher.vim/ftdetect/cypher.vim
012.186  000.006  000.006: sourcing /usr/share/vim/vimfiles/ftdetect/PKGBUILD.vim
012.200  000.006  000.006: sourcing /usr/share/vim/vimfiles/ftdetect/augeas.vim
012.229  000.022  000.022: sourcing /usr/share/vim/vimfiles/ftdetect/gnfiletype.vim
012.247  000.012  000.012: sourcing /usr/share/vim/vimfiles/ftdetect/meson.vim
012.372  004.106  004.034: sourcing /usr/share/nvim/runtime/filetype.vim
012.440  000.006  000.006: sourcing /usr/share/nvim/runtime/ftplugin.vim
012.496  000.005  000.005: sourcing /usr/share/nvim/runtime/indent.vim
012.620  000.045  000.045: sourcing /usr/share/nvim/runtime/syntax/synload.vim
012.662  000.146  000.101: sourcing /usr/share/nvim/runtime/syntax/syntax.vim
015.921  002.994  002.994: sourcing /home/sudokit/.config/nvim/plugged/tokyonight.nvim/colors/tokyonight-night.lua
423.544  000.478  000.478: sourcing /home/sudokit/.config/nvim/statusline.vim
423.553  419.214  410.648: sourcing /home/sudokit/.config/nvim/init.vim
423.557  000.300: sourcing vimrc file(s)
425.520  001.752  001.752: sourcing /home/sudokit/.config/nvim/plugged/vim-devicons/plugin/webdevicons.vim
425.610  000.030  000.030: sourcing /home/sudokit/.config/nvim/plugged/rainbow/plugin/rainbow_main.vim
425.761  000.082  000.082: sourcing /home/sudokit/.config/nvim/plugged/vim-MvVis/plugin/MvVis.vim
426.035  000.097  000.097: sourcing /home/sudokit/.config/nvim/plugged/coc.nvim/autoload/coc/rpc.vim
426.371  000.283  000.283: sourcing /home/sudokit/.config/nvim/plugged/coc.nvim/autoload/coc/util.vim
426.698  000.164  000.164: sourcing /home/sudokit/.config/nvim/plugged/coc.nvim/autoload/coc/client.vim
428.530  000.362  000.362: sourcing /home/sudokit/.config/nvim/plugged/coc.nvim/autoload/coc/highlight.vim
429.151  000.499  000.499: sourcing /home/sudokit/.config/nvim/plugged/coc.nvim/autoload/coc/color.vim
435.061  009.254  007.849: sourcing /home/sudokit/.config/nvim/plugged/coc.nvim/plugin/coc.vim
435.901  000.751  000.751: sourcing /home/sudokit/.config/nvim/plugged/fzf/plugin/fzf.vim
436.909  000.923  000.923: sourcing /home/sudokit/.config/nvim/plugged/fzf.vim/plugin/fzf.vim
437.031  000.043  000.043: sourcing /home/sudokit/.config/nvim/plugged/vim-snippets/plugin/vimsnippets.vim
437.462  000.353  000.353: sourcing /home/sudokit/.config/nvim/plugged/semshi/plugin/semshi.vim
437.726  000.182  000.182: sourcing /home/sudokit/.config/nvim/plugged/vim-commentary/plugin/commentary.vim
437.927  000.127  000.127: sourcing /home/sudokit/.config/nvim/plugged/vim-startify/plugin/startify.vim
439.415  001.414  001.414: sourcing /home/sudokit/.config/nvim/plugged/vim-fugitive/plugin/fugitive.vim
440.285  000.782  000.782: sourcing /home/sudokit/.config/nvim/plugged/vim-smoothie/plugin/smoothie.vim
440.455  000.088  000.088: sourcing /home/sudokit/.config/nvim/plugged/tmux-complete.vim/plugin/tmuxcomplete.vim
441.183  000.639  000.639: sourcing /home/sudokit/.config/nvim/plugged/vim-eunuch/plugin/eunuch.vim
441.824  000.411  000.411: sourcing /home/sudokit/.config/nvim/plugged/vim-sandwich/plugin/operator/sandwich.vim
442.251  000.404  000.404: sourcing /home/sudokit/.config/nvim/plugged/vim-sandwich/plugin/sandwich.vim
442.453  000.183  000.183: sourcing /home/sudokit/.config/nvim/plugged/vim-sandwich/plugin/textobj/sandwich.vim
442.603  000.073  000.073: sourcing /home/sudokit/.config/nvim/plugged/vim-tmux-navigator/plugin/tmux_navigator.vim
442.872  000.203  000.203: sourcing /home/sudokit/.config/nvim/plugged/markdown-preview.nvim/plugin/mkdp.vim
443.406  000.219  000.219: sourcing /usr/share/nvim/runtime/plugin/gzip.vim
443.436  000.011  000.011: sourcing /usr/share/nvim/runtime/plugin/health.vim
443.537  000.086  000.086: sourcing /usr/share/nvim/runtime/plugin/man.vim
444.278  000.292  000.292: sourcing /usr/share/nvim/runtime/pack/dist/opt/matchit/plugin/matchit.vim
444.388  000.835  000.544: sourcing /usr/share/nvim/runtime/plugin/matchit.vim
444.568  000.164  000.164: sourcing /usr/share/nvim/runtime/plugin/matchparen.vim
444.610  000.020  000.020: sourcing /usr/share/nvim/runtime/plugin/netrwPlugin.vim
444.781  000.012  000.012: sourcing /home/sudokit/.local/share/nvim/rplugin.vim
444.791  000.166  000.155: sourcing /usr/share/nvim/runtime/plugin/rplugin.vim
444.930  000.124  000.124: sourcing /usr/share/nvim/runtime/plugin/shada.vim
444.970  000.024  000.024: sourcing /usr/share/nvim/runtime/plugin/spellfile.vim
445.152  000.167  000.167: sourcing /usr/share/nvim/runtime/plugin/tarPlugin.vim
445.268  000.099  000.099: sourcing /usr/share/nvim/runtime/plugin/tohtml.vim
445.303  000.019  000.019: sourcing /usr/share/nvim/runtime/plugin/tutor.vim
445.532  000.214  000.214: sourcing /usr/share/nvim/runtime/plugin/zipPlugin.vim
445.648  000.019  000.019: sourcing /usr/share/vim/vimfiles/plugin/fzf.vim
445.781  000.115  000.115: sourcing /usr/share/vim/vimfiles/plugin/redact_pass.vim
446.715  003.180: loading rtp plugins
446.894  000.179: loading packages
448.004  000.476  000.476: sourcing /home/sudokit/.config/nvim/plugged/indentLine/after/plugin/indentLine.vim
448.083  000.713: loading after plugins
448.099  000.016: inits 3
448.373  000.274: reading ShaDa
449.006  000.633: opening buffers
449.061  000.055: BufEnter autocommands
449.063  000.002: editing files in windows
449.799  000.649  000.649: sourcing /home/sudokit/.config/nvim/plugged/vim-startify/autoload/startify.vim
865.216  410.551  410.551: sourcing /usr/share/nvim/runtime/autoload/provider/clipboard.vim
867.466  000.180  000.180: sourcing /home/sudokit/.config/nvim/plugged/vim-startify/syntax/startify.vim
868.082  000.158  000.158: sourcing /home/sudokit/.config/nvim/plugged/rainbow/autoload/rainbow_main.vim
868.355  000.092  000.092: sourcing /home/sudokit/.config/nvim/plugged/rainbow/autoload/rainbow.vim
871.970  011.277: VimEnter autocommands
871.975  000.005: UIEnter autocommands
871.979  000.004: before starting main loop
875.610  003.630: first screen update
875.613  000.004: --- NVIM STARTED ---`

**Comment by Blacksuan19** [comment]
the culprit is tokyo nights theme which is not part of this config.

```
015.921 002.994 002.994: sourcing /home/sudokit/.config/nvim/plugged/tokyonight.nvim/colors/tokyonight-night.lua
423.544 000.478 000.478: sourcing /home/sudokit/.config/nvim/statusline.vim
423.553 419.214 410.648: sourcing /home/sudokit/.config/nvim/init.vim
```
